nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -971,55 +959,56 @@ define(
                 });
 
                 fsm.bus.on('on-execute-requested', () => {
-                    ui.setContent('run-control-panel.execMessage', 'Sending...');
+                    setExecMessage('Sending...');
                 });
                 fsm.bus.on('exit-execute-requested', () => {
-                    ui.setContent('run-control-panel.execMessage', '');
+                    setExecMessage('');
                 });
 
                 fsm.bus.on('on-launched', () => {
-                    ui.setContent('run-control-panel.execMessage', 'Launching...');
+                    setExecMessage('Launching...');
                 });
                 fsm.bus.on('exit-launched', () => {
-                    ui.setContent('run-control-panel.execMessage', '');
+                    setExecMessage('');
                 });
 
                 fsm.bus.on('start-queueing', () => {
-                    doStartQueueing();
+                    updateJobState();
                 });
                 fsm.bus.on('start-running', () => {
-                    doStartRunning();
+                    updateJobState();
                 });
 
                 fsm.bus.on('on-completed', () => {
-                    doOnSuccess();
+                    updateJobState();
+                    generateJobOutput();
```

Comment:
This is the only FSM transition of any substance. I've split out the output cell creation stuff into a separate function, `generateJobOutput()`